test/powershell/Modules/Microsoft.PowerShell.Management/Set-Location.Tests.ps1

```diff
@@ -67,4 +67,29 @@ Describe "Set-Location" -Tags "CI" {
             (Get-Location).Path | Should -BeExactly (Get-PSProvider FileSystem).Home
         }
     }
+
+    It "Should set location to new drive's current working directory when path is the colon-terminated name of a different drive" {
+        try
+        {
+            Set-Location 'TestDrive:\'
```

Comment:
I believe we should save current location before set new one and restore in `finally`.

Reply:
Should I add that before the `try` block, or inside it?

Reply:
I meant:
```powershell
try {
    $oldLocation - Get-Location
    Set-Location 'TestDrive:\'
    ...
} finally {
    Set-Location $oldLocation
}
```

Reply:
Thanks, will add that now.